### PR TITLE
Standard Text Button

### LINKED ISSUE #69: Create button component
## Context
We will need a generic button component for creating buttons, adding backgrounds (if needed) to them, modifying their colour and size, and so on.

## Problem
We do not have such a component. A couple instances of the button component in the prototype are circled below:

Screenshot 2022-01-25 211814
 
As props, the button component should accept:
- An action to be executed on click (likely navigation)
- Text to be displayed
- If a background should be displayed
- Main and secondary colours
- Size
- If a shadow should be shown
- If an image should be displayed, and if so, what image. (e.g. the navigation arrow is an image but "sign up" is not)

## Solution
- [ ] Determine how this component should be implemented
- [ ] Implement it, making sure to generalize wherever possible
- [ ] Complete issue #70 for testing

### PR DESCRIPTION
**Pull request by DemiAdisa**
## Source 

Closes #69 

### Summary of PR 

Users will need to click buttons to perform actions. This PR intends to include a standard text button for our reuse throughout the app.

### Purpose 

To add a standard text button component to the app.

### Review and Testing Steps

- [ ] Pull this branch
- [ ] Read the StdButton Component documentation
- [ ] Run "npm install" in the client folder to install dependencies
- [ ] Run "npm start" and ensure you can see the default button on the screen
- [ ] Now head over to the Sign In Screen Component, a StdButton template has been included in the code, play around with all the props, and ensure they work as desired.
- [ ] Leave your review, do you think anything else needs to be added?

### Additional Information (If applicable)
Nil

### Notes
This PR does not include a test for the button.

### Resources 
Check [here](https://callstack.github.io/react-native-paper/button.html#icon) for resources used.

### CODE REVIEW

**Comment by sontushkpaul** [comment]
@DemiAdisa there is merge conflict in App.js file and please make sure not to add buttons in App.js file. you can add it in any screens from "screen" folder.

**Comment by DemiAdisa** [comment]
> @DemiAdisa there is merge conflict in App.js file and please make sure not to add buttons in App.js file. you can add it in any screens from "screen" folder.

I don't recall putting any buttons in the App.js file, let me go try and see what is going on

**Comment by DemiAdisa** [comment]
> @DemiAdisa there is merge conflict in App.js file and please make sure not to add buttons in App.js file. you can add it in any screens from "screen" folder.

Problem fixed, will update the PR testing instructions

**Review by ja-chisholm — APPROVED** [review]
The button functionality looks good, but a lot of files are modified here that don't seem related to the PR

File: app/client/mindjogg/components/StdButton/StdButton.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      onPress={props.buttonPress}
+    >
+      {props.text}
+    </Button>
+  );
+};
+
+StdButton.propTypes = {
+  uppercaseOn: PropTypes.bool.isRequired,
+  buttonIcon: PropTypes.string.isRequired,
```
Comment (sontushkpaul) [inline app/client/mindjogg/components/StdButton/StdButton.js]: @DemiAdisa Not all button have icon so how it is required?

File: app/client/mindjogg/components/StdButton/StdButton.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+This is a boolean value and is set to true by default. It sets the text to Uppercase when true.
+
+## buttonColour
+
+This will be a string (has to be a color hexcode e.g. #00008B) and when supplied a value, will set the background colur of the button.  
+By default, the prop is set to "#9B7FBA"
+
+## buttonPress
+This takes the function to be executed when the button is pressed.
+By default, the function is just a console.log("I do nothing").
```
Comment (sontushkpaul) [inline app/client/mindjogg/components/StdButton/StdButton.md:37]: @DemiAdisa Provide an example how to use it with all props

File: app/client/mindjogg/components/StdButton/StdButton.js (hunk 1 above)
Comment (DemiAdisa) [reply app/client/mindjogg/components/StdButton/StdButton.js]: Yes you are right, I just kind of wanted to just follow the type-scripting I was doing for the props so I just set it to required and then had it set to a default empty icon. This was more like a personal feeing I had while coding, I can update it if anyone disagrees with it

**Comment by sontushkpaul** [comment]
@DemiAdisa @jyaneshseeam  Please do not Merge with unresolved comments

File: app/client/mindjogg/components/StdButton/StdButton.md (hunk 2 above)
Comment (DemiAdisa) [reply app/client/mindjogg/components/StdButton/StdButton.md:37]: True, will include that shortly. Also, there is a console warning concerning the gesture used in the drawer navigation. Do not know if I am the only one who gets that warning on my device, if you do get it, lemme know.

**Comment by DemiAdisa** [comment]
> The button functionality looks good, but a lot of files are modified here that don't seem related to the PR

What files do you mean if I might ask?

**Comment by DemiAdisa** [comment]
> @DemiAdisa @jyaneshseeam Please do not Merge with unresolved comments

Noted👍

File: app/client/mindjogg/components/StdButton/StdButton.js (hunk 1 above)
Comment (sontushkpaul) [reply app/client/mindjogg/components/StdButton/StdButton.js]: did you check if it create any problem in UI or look bad or not?

File: app/client/mindjogg/components/StdButton/StdButton.md (hunk 2 above)
Comment (sontushkpaul) [reply app/client/mindjogg/components/StdButton/StdButton.md:37]: can you show me your error in the meeting? I am sure it should not happen

**Comment by ja-chisholm** [comment]
> > The button functionality looks good, but a lot of files are modified here that don't seem related to the PR
> 
> What files do you mean if I might ask?

Textbox is included in this PR (which is no big deal), but some backend stuff is as well 
image
which should go in their own PR imo

**Comment by DemiAdisa** [comment]
> > > The button functionality looks good, but a lot of files are modified here that don't seem related to the PR
> > 
> > 
> > What files do you mean if I might ask?
> 
> Textbox is included in this PR (which is no big deal), but some backend stuff is as well image which should go in their own PR imo

That is really weird, I did not even touch those files

**Comment by sontushkpaul** [comment]
> > > > The button functionality looks good, but a lot of files are modified here that don't seem related to the PR
> > > 
> > > 
> > > What files do you mean if I might ask?
> > 
> > 
> > Textbox is included in this PR (which is no big deal), but some backend stuff is as well image which should go in their own PR imo
> 
> That is really weird, I did not even touch those files

Let the merge other open PR first then merge yours. It should solve the problem

**Comment by DemiAdisa** [comment]
> > > > > The button functionality looks good, but a lot of files are modified here that don't seem related to the PR
> 
> > > > 
> 
> > > > 
> 
> > > > What files do you mean if I might ask?
> 
> > > 
> 
> > > 
> 
> > > Textbox is included in this PR (which is no big deal), but some backend stuff is as well image which should go in their own PR imo
> 
> > 
> 
> > That is really weird, I did not even touch those files
> 
> 
> 
> Let the merge other open PR first then merge yours. It should solve the problem 

Noted!

**Review by sontushkpaul — APPROVED** [review]
I tested and it worked.

**Pull request merged**